cmd/cadvisor.go

```diff
@@ -176,7 +230,23 @@ func main() {
 	rootMux.Handle(*urlBasePrefix+"/", http.StripPrefix(*urlBasePrefix, mux))
 
 	addr := fmt.Sprintf("%s:%d", *argIP, *argPort)
-	klog.Fatal(http.ListenAndServe(addr, rootMux))
+	listener, err := net.Listen("tcp", addr)
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	// Wrap in a TLS listener if cert/key was specfied
+	if *tlsCertPath != "" {
```

Comment:
I would rather check for `tlsConfig != nil` (assuming that it's of type `*tls.Config`).

Reply:
Done